packages/builder/src/components/nav/ModelSetupNav/ModelSetupNav.svelte

```diff
@@ -0,0 +1,127 @@
+<script>
+  import { getContext, onMount } from "svelte"
+  import { Button, Switcher } from "@budibase/bbui"
+  import { notifier } from "@beyonk/svelte-notifications"
+  import { store, backendUiStore } from "builderStore"
+  import api from "builderStore/api"
+  import ModelFieldEditor from "./ModelFieldEditor.svelte"
+
+  const { open, close } = getContext("simple-modal")
+
+  const ITEMS = [
+    {
+      title: "Setup",
+      key: "SETUP",
+    },
+    {
+      title: "Delete",
+      key: "DELETE",
+    },
+  ]
+
+  let edited = false
+
+  $: selectedTab = $backendUiStore.tabs.SETUP_PANEL
+
+  $: edited =
+    $backendUiStore.selectedField ||
+    ($backendUiStore.draftModel &&
+      $backendUiStore.draftModel.name !== $backendUiStore.selectedModel.name)
+
+  async function deleteModel() {
+    const model = $backendUiStore.selectedModel
+    const instanceId = $backendUiStore.selectedDatabase._id
+    const field = $backendUiStore.selectedField
+
+    if (field) {
+      delete model.schema[field]
+      backendUiStore.actions.models.save({ model, instanceId })
+      notifier.danger(`Field ${field} deleted.`)
+      return
+    }
+
+    const DELETE_MODEL_URL = `/api/${instanceId}/models/${model._id}/${model._rev}`
```

Comment:
do we need to think about cleaning up exiting records of this type, and the `all_<recordid>` view ?

Should we introduce a soft delete of records.... `budibase.isDeleted: true` and filter these out of all views? Too much?

Reply:
Same question (ish) to renaming of fields - do we try and clean this up? 

What happens if we 
1. Create model with `Size : string`
2. Create record with `"size":"small"`
3. Edit model with `Size: number` (or delete field and re-add)
4. Load previous record / view

Reply:
We do delete all the records and linked records for a model when we destroy it. Right now we don't support changing of types - only delete and re-add. I do think we should clean it up though because it will look weird if you delete a field then re-add it with the same name and all your data is there